byterun/custom.c

```diff
@@ -39,6 +39,12 @@ CAMLexport value caml_alloc_custom(struct custom_operations * ops,
     if (ops->finalize != NULL || mem != 0) {
       /* Remember that the block needs processing after minor GC. */
       add_to_custom_table (&caml_custom_table, result, mem, max);
+      /* Keep track of extra resources held by custom block in
+         minor heap. */
+      caml_extra_heap_resources_minor += (double) mem / (double) max;
```

Comment:
I am not knowledgeable enough to opine on the proposed strategy, but shouldn't we protect against invalid values of `mem` and `max` ? (e.g. `max == 0`)

Reply:
I've added some protection against `mem = 0` (just an optimization) and `max = 0`, setting it to 1 instead as `caml_adjust_gc_speed does`.  That function also truncates `mem` to be `<= max`, but this seems useless (it would be, strictly speaking, if the comparison for the threshold were `>= 1.` instead of `> 1`).  Note that both mem and max are unsigned integer, so no need to deal with negative values.

Reply:
> max = 0, setting it to 1

Actually, this might be useless as well (just let the ratio be +infinity, which would trigger the minor gc immediately; this is also the case if max = 1, mem > 0 anyway).